pkgs/by-name/jd/jdt-language-server/package.nix

```diff
@@ -2,11 +2,12 @@
 , stdenv
 , fetchurl
 , python3
-, jdk
+, jdk17
```

Comment:
How about overriding in `pkgs/top-level/all-packages.nix`?

https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name#changing-implicit-attribute-defaults

Reply:
@wegank It would be better if you responded to the discussion about the general approach rather than this minor detail. 

By the way, hardcoding JDK 17 will break this package for users who are working on JDK 19. I believe generally we want to use the highest supported version.